lua/neo-tree/setup/init.lua

```diff
@@ -71,6 +71,11 @@ local define_events = function()
   events.define_autocmd_event(events.VIM_COLORSCHEME, { "ColorScheme" }, 0)
   events.define_autocmd_event(events.VIM_CURSOR_MOVED, { "CursorMoved" }, 100)
   events.define_autocmd_event(events.VIM_AFTER_SESSION_LOAD, { "SessionLoadPost" }, 200)
+  events.define_autocmd_event(
+    events.VIM_LSP_UPDATE,
```

Comment:
The convention is that events that start with `VIM_` are direct mappings to those built-in events. Multiple events are only combined in special situations where their meanings are synonymous for our purposes.

You should just be adding the individual events that do not already exist, rather than creating one super event.

Reply:
@cseickel I think it is better to bundle these events into one `autocmd`, so that they have a unified debouncing time. This will limit the number of calls to the LSP server, which could slow down nvim. I'll name it `NEO_TREE_LSP_UPDATE` if that is fine for you.

Reply:
Actually, the debounce can happen on the subscriber side instead of of on the producer (vim events) side. The important thing here is that we don't want to subscribe to the same autocommand ore then once, that's why one at a time is the way to go.

The function the end up calling should have it's own debounce if you worried about multiple vim events happening at the same time. That is how the filesystem refresh works.

Reply:
Currently, I'm switching back to the `LspRequest` event for auto-refresh. It's quite fast with a debounce time of 200ms, and covers many events, so I think it is a good idea for now.